modules/renter/contractor/contractor.go

```diff
@@ -133,6 +134,7 @@ func newContractor(cs consensusSet, w wallet, tp transactionPool, hdb hostDB, p
 
 		cachedRevisions: make(map[types.FileContractID]cachedRevision),
 		contracts:       make(map[types.FileContractID]modules.RenterContract),
+		downloaders:     make(map[types.FileContractID]*hostDownloader),
```

Comment:
What happens if the file contract id changes due to a renew? Host key might be a better key, but then if you're basically limited to exactly 1 contract per host, which might be limiting but ultimately I think we could work around any problems that it might introduce.

At some point in the future we're going to want to be able to use the funds from previous contracts (renewed contracts can still have money in them), but that seems like something the contractor should be worrying about, and not the renter. Since this is just an ID for the renter, it's probably okay to limit the renter to seeing one element per host. Especially since it's just a pointer and the renter is allowed to have multiple of those elements out at a time.